### PR TITLE
[Keyboard] Use old custom matrix for Drop Planck+Preonic

### LINKED ISSUE #8610: [Bug] Matrix scan issues on Planck Rev 6 and Rev 6.1 on versions 0.7.6 and up
## Describe the Bug

Pressing certain keys on Planck Rev 6 and Rev 6.1 results in multiple presses being registered, as if the board had ghosting issues. This is not a hardware/electrical issue since it works fine on older versions of QMK.

For example, pressing Enter on the default keymap outputs `tg'`. It seems to be a matrix scanning issue.

image

After some investigation and testing, I've determined that the broken behavior was brought about by #6140. This PR removed the custom `matrix.c` files for Planck Rev 3 and Rev 6 since the DIP switch functionality contained therein was made a core feature. However, this apparently also messed with the scan/debounce timing (or something of the sort), which resulted in broken matrix scanning.

## System Information

 - Keyboard: Planck
   - Revision: Rev 6, Rev 6.1
 - Operating system: Windows
 - ARM GCC version: 6.3.1 20170620
 - QMK Firmware version: 0.7.6 and up

## Additional Context

This issue was also experienced by the OP of [this thread](https://www.reddit.com/r/olkb/comments/earda2/problem_flashing_new_planck_with_qmk_toolbox/) on the OLKB subreddit. Setting `DIP_SWITCH_ENABLE = no` as @drashna suggested in the thread didn't fix the problem.

Other affected boards may include Planck Rev 3 and Preonic Rev 3.

### PR DESCRIPTION
**Pull request by drashna**
## Description

There have been numerous reports of the Drop OLKB boards having issues which are fixed by reverting to the old custom matrix.  Since it seems to be limited to Drop issues boards, added a new revision just for them.

Did not use custom matrix, as wanted to change as little as possible. 

## Types of Changes

- [x] Bugfix
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)

## Issues Fixed or Closed by This PR

* Reported issues on discord, fixes #8610 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/preonic/drop/rules.mk [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
+
+# Do not enable RGB_MATRIX_ENABLE together with RGBLIGHT_ENABLE
+RGB_MATRIX_ENABLE = no
+RGB_MATRIX_DRIVER = WS2812
+
+ENCODER_ENABLE = yes
+DIP_SWITCH_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/preonic/drop/rules.mk:33]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = yes          # Audio output
WS2812_DRIVER = pwm
CUSTOM_MATRIX = yes
ENCODER_ENABLE = yes
DIP_SWITCH_ENABLE = yes

# Do not enable RGB_MATRIX_ENABLE together with RGBLIGHT_ENABLE
RGB_MATRIX_ENABLE = no
RGB_MATRIX_DRIVER = WS2812

SRC += matrix.c
```

File: keyboards/planck/drop/readme.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Planck
+
+A compact 40% (12x4) ortholinear keyboard kit made and sold by OLKB and Massdrop. A complete hardware rework of the Planck, sporting a faster and more powerful STM32 ARM Cortex-M4 microcontroller, with support for rotary encoders and three additional layouts. [More info on qmk.fm](http://qmk.fm/planck/)
+
+Keyboard Maintainer: [Jack Humbert](https://github.com/jackhumbert)  
+Hardware Supported: Planck PCB rev6  
+Hardware Availability: [OLKB.com](https://olkb.com), [Massdrop](https://www.massdrop.com/buy/planck-mechanical-keyboard?mode=guest_open)
```
Comment (fauxpark) [inline keyboards/planck/drop/readme.md:7]:
```suggestion
* Keyboard Maintainer: [Jack Humbert](https://github.com/jackhumbert)
* Hardware Supported: Planck PCB rev6
* Hardware Availability: [OLKB.com](https://olkb.com), [Massdrop](https://www.massdrop.com/buy/planck-mechanical-keyboard?mode=guest_open)
```

File: keyboards/planck/drop/rules.mk [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+SRC += matrix.c
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
+
+RGB_MATRIX_ENABLE = no
+RGB_MATRIX_DRIVER = WS2812
+ENCODER_ENABLE = yes
+DIP_SWITCH_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/planck/drop/rules.mk:31]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = yes          # Audio output
WS2812_DRIVER = pwm
CUSTOM_MATRIX = yes
# Do not enable RGB_MATRIX_ENABLE together with RGBLIGHT_ENABLE
RGB_MATRIX_ENABLE = no
RGB_MATRIX_DRIVER = WS2812
ENCODER_ENABLE = yes
DIP_SWITCH_ENABLE = yes

SRC += matrix.c
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by jackhumbert** [comment]
Here's the two bin files (in .zip archives) of the default keymap to test if this addresses the issue - they can be flashed with QMK Toolbox:

* [preonic_rev3_drop_default](https://github.com/qmk/qmk_firmware/files/7190786/preonic_rev3_drop_default.zip)
* [planck_rev6_drop_default](https://github.com/qmk/qmk_firmware/files/7190787/planck_rev6_drop_default.zip)

**Review by jackhumbert — APPROVED** [review]

**Comment by samjonester** [comment]
Thanks for this, @drashna! When checking out this PR I've had success flashing both default, and custom firmware with a planck rev6 purchased from drop in early September 2021. Keyboard tester shows everything working.

``` sh
qmk compile -kb planck/rev6_drop -km default
qmk flash -kb planck/rev6_drop -km default
```

**Pull request merged**